automotiveMastermind/condo
# feat(node): add node build/test support

* serialize package.json to get node metadata
* serialize bower.json to get bower metadata
* add support for node scripts
    * build - will execute if a node script called "condo" does not exist
    * test - will execute if a node script called "condo" does not exist
    * condo - executes in place of build / test -- it is assumed that
      this script will call all other targets required for ci loop
* add support for additional polymer tasks
    * lint
    * test
* update polymer-cli and all dependencies to latest in frontend test
* add skip support by technology stack
    * `SKIP_NODE` - skips all node related targets / tasks
    * `SKIP_DOTNET` - skips all dotnet related targets / tasks
    * `SKIP_BOWER` - skips all bower related targets / tasks
    * `SKIP_POLYMER` - skips all poylmer related targets / tasks
* add the following skip variables
    * `SKIP_POLYMER_LINT` - skips the lint target for polymer
    * `SKIP_POLYMER_BUILD` - skips the build target for polymer
    * `SKIP_POLYMER_TEST` - skips the test target for polymer
* local build no longer skips frontend technology stacks by default
    * removed --local from condo-cli
    * update circleci set skip environment variables instead

Closes: #107
